File: data_formatting.R
library(tidyverse)

#--------DATA FORMATTING----------#
filename <- list.files(pattern = '.csv')

data <- lapply(filename, read.csv)

for (i in 1:length(data)){
  data[[i]] <- cbind(data[[i]],'year' = substr(filename[i],17,20))
  }

df <- do.call(rbind, data)

colnames(df) <- tolower(colnames(df))

#create county comission district
df <- df %>% mutate(ccd = as.factor(if_else(df$precinctcode < 1000, substr(df$precinctcode,1,1), substr(df$precinctcode,1,2))))

#create school board district
df <- df %>% mutate(sbd = case_when(ccd %in% (1:2) ~ 1,
                                    ccd %in% (3:4) ~ 2,
                                    ccd %in% (5:6) ~ 3,
                                    ccd %in% (7:8) ~ 4,
                                    ccd %in% (9:10) ~ 5,
                                    ccd %in% (11:12) ~ 6,
                                    ccd %in% (13:14) ~ 7
                                   )
                   )

df$party <- as.character(df$party)
df$candidate_issue <- as.character(df$candidate_issue)

df$party <- if_else(df$party %in% c('DEM','REP'),df$party,'OTH')
df$candidate_issue <- if_else(df$party %in% c('DEM','REP'),df$candidate_issue, 'OTH')

df$contest <- as.character(df$contest)

df$contest[df$contest == "GOVERNOR AND LIEUTENANT GOVERNOR"] <- "Governor"
df$contest[df$contest == "MAYOR"] <- "Mayoral"
df$contest[df$contest == "United States President and Vice President"] <- "Presidental"

df <-  df %>% filter(contest %in% c("Governor", "Mayoral",
                                    "Presidental")) %>% 
              select(year,party, contest,ccd,sbd, candidate_issue, totalvotes)


#combine year and contest for unique election type
df$election <- paste(df$year,df$contest,"Election")

write.csv(df, "df.csv", row.names=FALSE)

File: data_formatting.py
import glob
import os
import pandas as pd
import numpy as np

filename = glob.glob("*.csv")

df = pd.concat([pd.read_csv(i).assign(Year = os.path.basename(i)[16:20])  
                for i in filename],ignore_index = True)

#convert column names to lowercase
df.columns = map(str.lower, df.columns)

#create county comission district
def ccdcat(x):
    if x < 1000:
        return str(x)[:1]
    else:
        return str(x)[:2]

df['ccd'] = df['precinctcode'].apply(ccdcat)

#create school board district
def sbdcat(x):
    if x in ('1','2'):
        return '1'
    elif x in ('3','4'):
        return '2'
    elif x in ('5','6'):
        return '3'
    elif x in ('7','8'):
        return '4'
    elif x in ('9','10'):
        return '5'
    elif x in ('11','12'):
        return '6'
    elif x in ('13','14'):
        return '7'

df['sbd'] = df['ccd'].apply(sbdcat)


df['party'] = np.where(df.party.isin(['DEM','REP']), df.party,'OTH')
df['candidate_issue'] = np.where(df.party.isin(['DEM','REP']), df.candidate_issue,'OTH')

df.contest.replace(["GOVERNOR AND LIEUTENANT GOVERNOR","MAYOR","United States President and Vice President"],
                   ["Governor","Mayoral","Presidental"], inplace=True)

df = df.query('contest == "Governor" | contest == "Presidental" \
               |contest == "Mayoral"')

df = df[['year','party', 'contest','ccd','sbd', 'candidate_issue', 'totalvotes']]

#combine year and contest for unique election type
df['election'] = df['year'] + " " + df['contest'] + " " + "Election"

df.to_csv("df.csv", index = False)

